algebird-core/src/main/scala/com/twitter/algebird/Monoid.scala

```diff
@@ -48,12 +48,8 @@ trait Monoid[@specialized(Int, Long, Float, Double) T]
       throw new java.lang.IllegalArgumentException("argument should not be zero")
     }
 
-  def nonZeroOption(v: T): Option[T] =
-    if (isNonZero(v)) {
-      Some(v)
-    } else {
-      None
-    }
+  def nonZeroOption(v: T): Option[T] = Some(v).filter(isNonZero)
```

Comment:
This requires allocating a closure and virtual dispatch that won’t be inlined. I’d rather not change this since it can almost only reduce performance.

Reply:
Hi @johnynek thanks for looking at this! oh i see it's a good point; i'll revert this! 👍

Reply:
Ps: I hope that scala 3 can fix this by making Option filter map etc... an in-line def.

Reply:
+1 on that